### DISCUSSION TITLE
Roadmap 2022

### DISCUSSION THREAD

**Discussion by evpobr** [libsndfile/libsndfile#701] [General]
- [x] New minor release (#692)
- [x] Replace `CHANGELOG` with `CHANGELOG.md`
- [x] MP3 support (#499)
- [ ] WavPack support (#661)
- [ ] New open function

## New open function

```c
int sf_open_path(const char *name, int mode, SF_INFO *sfinfo, SF_OPEN_CONFIG *sfconf, SNDFILE **sndfile);
```

Returns error, uses no static variables.

```c
struct SF_OPEN_CONFIG
{
    int struct_size;
    int flags;
    ...
};
```

Flags:

* `SF_OPEN_UTF8_PATH` - path is UTF8 for Windows platform.

**Comment by oplatek** [comment]
Hi!
Is the MP3 supported now since  #499 is closed?

I am interested in the MP3 in https://github.com/bastibe/python-soundfile/issues/291 and this seems to be the blocker.

Thanks for the info!

**Reply by evpobr** [reply]
Hi @oplatek . Yes, thanks to @arthurt it is supported now.

**Comment by evpobr** [comment]
@arthurt , @SoapGentoo , let's discuss the new release again.

It bothers me that trying to do a good thing, i mean polishing the MP3 to a shine, we start doing a bad thing, that is, leaving people without a release at all.

**Reply by SoapGentoo** [reply]
Agreed, when do you want to cut the release?

**Reply by evpobr** [reply]
We need to ask @arthurt too.

**Reply by arthurt** [reply]
Yes, I agree, we should cut a release soon. Sorry, I think I have been the one holding this back with a "... and one more thing", but not finding the time to do it.

I would like to fix #794 soon, but maybe that could be 1.1.1. Also, there's an issue with Vorbis clipping and normalization that should be addressed. Again, that should wait

**Reply by evpobr** [reply]
@arthurt , let's do it in next releases. @SoapGentoo , what about the release this weekend?

**Reply by SoapGentoo** [reply]
sure, can do

**Reply by evpobr** [reply]
Ok, will start release branch tomorrow.

**Reply by evpobr** [reply]
@SoapGentoo , https://github.com/libsndfile/libsndfile/tree/release-1.1.0

Release when it's convenient for you, I'll add Window binaries.

**Reply by evpobr** [reply]
@SoapGentoo , are you busy?

**Reply by SoapGentoo** [reply]
sorry, totally swamped in work, let me do it next weekend?

**Reply by evpobr** [reply]
No problem.

**Reply by SoapGentoo** [reply]
working on it now

**Reply by SoapGentoo** [reply]
@evpobr release is up

**Reply by evpobr** [reply]
Great, tank you.

**Comment by evpobr** [comment]
@SoapGentoo , WIn32 binaries:
[libsndfile-1.1.0-win64.zip](https://github.com/libsndfile/libsndfile/files/8357685/libsndfile-1.1.0-win64.zip)
[libsndfile-1.1.0-win32.zip](https://github.com/libsndfile/libsndfile/files/8357687/libsndfile-1.1.0-win32.zip)

**Reply by SoapGentoo** [reply]
uploaded

**Comment by SoapGentoo** [comment]
@evpobr looking at `configure.ac` I was wondering whether we should finally abandon the non-fixed integers? Let's use `int32_t` instead of `int` and [avoid all of these pointless checks](https://github.com/libsndfile/libsndfile/blob/master/configure.ac#L187). Relying on `int64_t` is so much nicer than all of these native `long` etc types that depend on ILP64 etc.

**Comment by evpobr** [comment]
I agree. Although if we do it in one commit, we will rewrite most of the library :smile:

**Comment by evpobr** [comment]
@SoapGentoo , @arthurt , what about new release?

**Reply by SoapGentoo** [reply]
sure, we can cut one

**Reply by evpobr** [reply]
Ok. Let's wait for @arthurt .

**Reply by arthurt** [reply]
We should get the ~~wav_subchunk_parse() #873 fix in.~~ done

Also:
 - #830 has a pretty small one line fix.
 - #863 has a one line fix.

Can anyone else help with #873? I need to get a more iterative-friendly local fuzzer setup. I'm drawing a blank why the fuzzer is killing the changes for that.

**Reply by evpobr** [reply]
@AdamKorcz , @cmeister2 , can you help us please?

**Reply by cmeister2** [reply]
I'll take a look.

**Reply by evpobr** [reply]
Thanks 👍

**Reply by cmeister2** [reply]
Took a look, couldn't repro using the artifact crash test case and the arthurt branch. Might be a random false positive. If it goes away on a retry, I'd chalk it up to some oss-fuzz weirdness.

**Reply by evpobr** [reply]
@arthurt , I don't see any reproducible error anymore.
